pkg/devfile/parser/writer.go

```diff
@@ -11,6 +14,11 @@ import (
 // WriteYamlDevfile creates a devfile.yaml file
 func (d *DevfileObj) WriteYamlDevfile() error {
 
+	// Check kubernetes components, and restore original uri content
+	err := restoreK8sCompURI(d)
```

Comment:
just a thought:
this func would be checking for the attribute every time it is called to write devfile. I wonder if we should introduce some property in the object that we can use to keep track:

```

type DevfileObj struct {

	// Ctx has devfile context info
	Ctx devfileCtx.DevfileCtx

	// Data has the devfile data
	Data data.DevfileData

        // convert
        ConvertUriToInlined bool
}
```

Reply:
I do not want the value to be direct accessible.. so added into `DevfileObj.ctx.convertUriToInlined` instead.

Reply:
Thinking if we should really keep this context? Because context refers to a devfile context but otherwise its ok i guess